src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,211 @@
+                Statement getAllStatement = connection.createStatement()) {
+            ResultSet resultSet =
+                    getAllStatement.executeQuery(getAllRequest);
+            while (resultSet.next()) {
+                allCars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all cars from DB", e);
+        }
+        allCars.forEach(car -> car.setDrivers(getAllDriversByCarId(car.getId())));
+        return allCars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars SET model = ?, "
+                + "manufacturer_id = ? WHERE id = ? and is_deleted = false;";
+        int updateCount;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement =
+                        connection.prepareStatement(updateRequest)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateCount = updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update car by id: "
+                    + car.getId() + " from DB", e);
+        }
+        if (updateCount > 0) {
+            deleteDrivers(car);
+            insertDrivers(car);
+            return car;
+        }
+        throw new DataProcessingException("Car with id:" + car.getId() + "wasn`t found in DB");
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE cars SET is_deleted = true where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement =
+                        connection.prepareStatement(deleteRequest)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete car by id: "
+                    + id + " from DB", e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car =
+                new Car(resultSet.getString("model"),
+                        manufacturerService.get(resultSet
+                                .getObject("manufacturer_id", Long.class)));
+        car.setId(resultSet.getObject("id", Long.class));
+        return car;
+    }
+
+    public List<Car> getCarsByDriverId(Long driverId) {
+        String getCarsRequest =
+                "SELECT c.id AS id, c.model AS model, m.id AS manufacturer_id FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "WHERE c.is_deleted = FALSE AND cd.driver_id = ?;";
```

Comment:
let's add where condition to skip removed drivers as well

Reply:
added